front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -376,6 +396,44 @@ const handleNodeOperation = ({
   }
 };
 
+const handleLabelOperation = async ({
+  type,
+  label,
+  netzgrafikDto,
+  dispatch,
+  infraId,
+  addUpsertedTrainSchedules,
+}: {
+  type: NGEEvent['type'];
+  label: Label;
+  netzgrafikDto: NetzgrafikDto;
+  dispatch: AppDispatch;
+  infraId: number;
+  addUpsertedTrainSchedules: (trainSchedules: TrainScheduleResult[]) => void;
+}) => {
+  const { trainruns } = netzgrafikDto;
+  switch (type) {
+    case 'update': {
+      const trainrunUpdateLabels = trainruns.filter((trainrun) =>
+        trainrun.labelIds.includes(label.id)
+      );
+      trainrunUpdateLabels.forEach(async (trainrun) => {
+        await handleUpdateTrainSchedule({
+          netzgrafikDto,
+          trainrun,
+          dispatch,
+          infraId,
+          addUpsertedTrainSchedules,
+        });
+      });
+      break;
+    }
+    default: {
+      break;
+    }
+  }
```

Comment:
an `if` seems enough, do we wait for more cases?

Reply:
For maybe other cases in the future and harmonized with other oparation handle.